src/main/kotlin/com/theapache64/stackzy/util/CommandExecutor.kt

```diff
@@ -40,22 +40,22 @@ object CommandExecutor {
         val stdError = BufferedReader(InputStreamReader(proc.errorStream))
 
         // Read the output from the command
-        var s: String?
+        var lastLine: String?
         val result = mutableListOf<String>()
-        while (stdInput.readLine().also { s = it } != null) {
+        stdInput.forEachLine { line ->
             if (isLivePrint) {
-                println(s)
+                println(line)
             }
-            result.add(s!!)
+            result.add(line)
         }
 
         // Read any errors from the attempted command
         val error = StringBuilder()
-        while (stdError.readLine().also { s = it } != null) {
+        stdError.forEachLine { line ->
             if (isLivePrint) {
-                println(s)
+                println(line)
             }
-            error.append(s).append("\n")
+            error.append(line).append("\n")
         }
```

Comment:
that's better

Reply:
btw. there's even a different method to get all lines directly. in case that's what you'd prefer.

Reply:
nope, I need line by line. Commands like `apk-tool` take a lot of time to finish their execution. so `livePrint` won't work then.